sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/BucketMapper.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.aggregator;
+
+/**
+ * Interface defining methods for mapping values to buckets for {@link
+ * io.opentelemetry.sdk.metrics.data.ExponentialHistogramBuckets}.
+ */
+interface BucketMapper {
```

Comment:
True, there are reasons to have up to two implementations that we know about. The logarithm implementation stands in as a single implementation that works for all scales, but there are faster implementations if you restrict scale to non-negative values and non-positive values. It might be prudent to leave this an implementation detail until it comes to adding a second -- when I measured the difference between the logarithm method and the lookup table it was very slight, around 10ns difference. Is it worth having several more classes loaded to save 10ns per histogram observation? (Not sure.)

To those wondering why negative scale values matter, since they are indeed "extreme": for an auto-scaling histogram like our several prototypes, I believe it is easier to allow scale to go negative as a natural consequence of extreme input-value range than to check for and defend against extreme values at various other locations in the code.

Reply:
@jmacd The logarithm issue for me is less about the speed and more about the edge case where it returns an index that's off by one. I saw this when testing `Double.MIN_VALUE` and `Double.MAX_VALUE` in the same histogram, but it can also happen in more cases where the recorded value is close to a power of two. 

Some more discussion on that elsewhere in this PR: https://github.com/open-telemetry/opentelemetry-java/pull/3724#issuecomment-947870925